main.go

```diff
@@ -35,10 +34,10 @@ func checkForErrorsAndExit(err error) {
 			logger.Println(err)
 		}
 		// exit with the underlying error code
-		var retCode int = 1
-		if exiterr, ok := errors.Unwrap(err).(*exec.ExitError); ok {
-			status := exiterr.Sys().(syscall.WaitStatus)
-			retCode = status.ExitStatus()
+		var retCode int
+		retCode, err := shell.GetExitCode(err)
+		if err != nil {
+			retCode = 1
```

Comment:
`GetExitCode` will just return the error it was passed if an exit code can't be determined. That error will be  logged on line 34. It does make sense to log something, perhaps something like 

```go
logger.Println("Unable to determine underlying exit code, so Terragrunt will exit with error code 1")
```

would be more appropriate so that the logs aren't redundant. I know for sure that calling `terragrunt` from a location that doesn't have a config file satisfies this condition (i.e., terragrunt has to be coerced to exit with a non-zero error code).